### ISSUE TITLE
[BUG] Access denied on Windows when attempting to delete a file created by Cython during the build

### ISSUE THREAD

**Issue by evhub** [cython/cython#4489]
**Describe the bug**
When compiling on Windows, Cython seems to create a read-only file and then attempt to delete it, always resulting in an access denied error even when run from an administrator shell.

This is a new issue that seems to have started relatively recently without any changes in my build process. It also only appears on Windows, everything worked fine when I was testing on [WSL](https://docs.microsoft.com/en-us/windows/wsl/).

Here's the relevant part of the build log:
```
    Generating code
    Finished generating code
    copying build\lib.win-amd64-3.9\cPyparsing.cp39-win_amd64.pyd ->
    error: could not delete 'cPyparsing.cp39-win_amd64.pyd': Access is denied
```
(and [here](https://gist.github.com/evhub/dd854a9833204ea2fc3c79325969dc82) is a gist with the full log)

The file it is attempting to delete here is one that it previously created in my `build` directory, and this error persists even when run as administrator. It's entirely possible I'm just doing something dumb, but if so I'm not sure what it could be.

**To Reproduce**
The Cython project I've been experiencing this error with is [`cPyparsing`](https://github.com/evhub/cpyparsing). It should be reproducible if you clone the repository and `pip install -e .` on Windows.

**Expected behavior**
At least when run as administrator, I shouldn't be seeing access denied errors.

**Environment (please complete the following information):**
 - OS: Windows
 - Python version: 3.9.7
 - Cython version: 0.29.24

**Comment by scoder** [comment]
Thanks for the report. Cython doesn't actually build the `.pyd` file itself, that's up to distutils/setuptools.

Can't say what's going wrong here right now, but it seems funny that it says "copying bla.pyd -> [NOTHING]". Shouldn't there be a file name in the end?

**Comment by proptrader33** [comment]
I just discovered that the part that follows `->` is the relative path to which the file is to be copied. So `-> [nothing]` simply means the file is being moved to the current directory, since the relative path of the current directory is represented by an empty string, just as you'd see `-> folder1\folder2` in case of the .pyd file being copied to `folder1\folder2` where folder1 is a folder in the current directory.

**Comment by FI-Mihej** [comment]
Working dir cleanup (from temporary files) fixes this issue. I am using next sequence of git commands:
```
git reset
git checkout .
git clean -fdx
```
(do not forget to commit or stash any valuable changes before these commands execution)

**Comment by FI-Mihej** [comment]
Also all instancess of target Python interpreter must be finished before cleanup (see my comment above) and complilation. This means that you must finish you VSCode, PyCharm or other IDE as well.

The issue source seems to be in a difference between Linux and Windows approach: in Linux you are allowed to remove file of launched application or loaded .so file however you are not allowed to do the same in same situation on Windows.

Bit later I'll think what we can do about this. Unload .dll from target Python instance whould be perfect if possible. But script for reloading Python instance might be enough for a development purpose.
